meshroom/ui/reconstruction.py

```diff
@@ -572,8 +572,20 @@ def updateCameraInits(self):
         if set(self._cameraInits.objectList()) == set(cameraInits):
             return
         self._cameraInits.setObjectList(cameraInits)
+
+        # Save index of the active CameraInit and check if it still exists in the
+        # updated list
+        idx = -1
+        if self.cameraInit in cameraInits and self.getCameraInitIndex() != -1:
+            idx = self.getCameraInitIndex()
+
+        # Reset the index with the first CameraInit in the list
         self.cameraInit = cameraInits[0] if cameraInits else None
 
+        # If the previously active CameraInit node still exists, set it back as the active one
+        if idx != -1 and self.cameraInit is not None:
+            self.setCameraInitIndex(idx)
+
```

Comment:
Maybe simpler to update the CameraInit only when we need it (first assignment or when the old cameraInit is no more part of the scene, like when we load another project):
```suggestion
        if self.cameraInit is None or self.cameraInit not in cameraInits:
            self.cameraInit = cameraInits[0] if cameraInits else None
```

Reply:
I tested that solution and it does not work.

When the list of `CameraInits` is updated, the model from the combobox is emptied and refilled, and if a "cameraInitIndexChanged" signal is not emitted by actively setting `self.cameraInit`, no further `setCameraInitIndex` is effective as the corresponding model does not exist anymore. We need to "transition" by the first `CameraInit` in the list to keep the correct index when the list is updated.

Reply:
We could emit 'cameraInitChanged' manually, isn't it?
In all case, we could avoid using the index here and reset self.cameraInit directly.

Reply:
Done:
- the active `CameraInit` is only modified if it hasn't been assigned yet or if it has been removed from the list
- the `cameraInitChanged()` signal is emitted every single time the list of `CameraInits` is modified and it is connected to the QML to ensure that the combo box index, which is reset every single time the list is modified no matter what, is re-updated with the correct value